srv/modules/runners/ui_rgw.py

```diff
@@ -0,0 +1,122 @@
+                                'secret_key': "0123456789012345678901234567890123456789",
+                                'urls': ["http://red1",
+                                         "http://red2",
+                                         "http://blue1:8000",
+                                         "http://blue2:8000"]}
+
+    def _admin(self, filename="user.admin.json"):
+        """
+        Expect admin user file; otherwise, search for first system user.
+        Update access_key, secret_key
+        """
+        filepath = "{}/{}".format(self.pathname, filename)
+        if os.path.exists(filepath):
+            user = json.loads(open(filepath).read())
+        else:
+            user = None
+            for user_file in glob.glob("{}/user.*".format(self.pathname)):
+                user = json.loads(open(user_file).read())
+                if 'system' in user and user['system'] == "true":
+                    break
+                user = None
+            if not user:
+                # No system user
+                log.error("No system user for radosgw found")
+                return
+        self.credentials['access_key'] = user['keys'][0]['access_key']
+        self.credentials['secret_key'] = user['keys'][0]['secret_key']
+
+    def _urls(self):
+        """
+        Check for user defined endpoint; otherwise, return list of gateways as
+        urls.
+        """
+        search = "I@cluster:{}".format(self.cluster)
+        pillar_util = salt.utils.master.MasterPillarUtil(search, "compound",
+                                                         use_cached_grains=True,
+                                                         grains_fallback=False,
+                                                         opts='__opts__')
+        cached = pillar_util.get_minion_pillar()
+        for minion in cached:
+            if 'rgw_endpoint' in cached[minion]:
+                self.credentials['urls'].append(cached[minion]['rgw_endpoint'])
+                return
+
+        for client_file in glob.glob("{}/client.*".format(self.pathname)):
+            port = None
+            with open(client_file, 'r') as rgw_file:
+                for line in rgw_file:
+                    if 'port=' in line:
+                        match = re.search(r'port=(\d+)', line)
+                        port = match.group(1)
+            parts = client_file.split('.')
+            resource = None
+            # dedicated keys - use host part
+            if len(parts) == 4:
+                resource = parts[2]
+            # shared keys - use role part
+            if len(parts) == 3:
+                resource = parts[1]
+            if port:
+                resource += ":{}".format(port)
+            if resource:
+                self.credentials['urls'].append("http://{}".format(resource))
+
+
+def credentials(canned=None):
+    """
+    Return the administrative credentials for the RadosGW
+    """
+    radosgw = Radosgw(int(canned))
```

Comment:
A try/catch would be good idea... here or in Radosgw:__init__().  What about changing the default of 'canned' from None to 0 across the patch as well?